R/tm_g_patient_profile.R

```diff
@@ -424,647 +402,538 @@ srv_g_patient_profile <- function(id,
+  if (!is.na(rs_dataname)) checkmate::assert_names(rs_dataname, subset.of = names(data))
+  if (!is.na(lb_dataname)) checkmate::assert_names(lb_dataname, subset.of = names(data))
+  if (!is.na(cm_dataname)) checkmate::assert_names(cm_dataname, subset.of = names(data))
+  checkboxes <- c(ex_dataname, ae_dataname, rs_dataname, lb_dataname, cm_dataname)
   moduleServer(id, function(input, output, session) {
-    iv <- shinyvalidate::InputValidator$new()
-    iv$add_rule("sl_start_date", shinyvalidate::sv_required())
-    iv$add_rule("lb_var_show", shinyvalidate::sv_required())
-    iv$add_rule("ae_var", shinyvalidate::sv_required())
-    iv$enable()
+    select_plot <- reactive(
+      vapply(checkboxes, function(x) x %in% input$select_ADaM, logical(1L))
+    )
 
-    # only show the check box when domain data is available
-    observeEvent(ae_dataname, {
-      if (!is.na(ae_dataname)) {
-        output$select_ae_output <- renderUI({
-          checkboxInput(
-            session$ns("select_ae"),
-            "ADAE",
-            value = !is.na(ae_dataname)
-          )
-        })
-      }
-    })
+    if (!is.na(lb_dataname)) {
+      observeEvent(input$lb_var, ignoreNULL = TRUE, {
+        ADLB <- data[[lb_dataname]]() # nolint
+        choices <- unique(ADLB[[input$lb_var]])
+        choices_selected <- if (length(choices) > 5) choices[1:5] else choices
+
+        updateSelectInput(
+          session,
+          "lb_var_show",
+          selected = choices_selected,
+          choices = choices
+        )
+      })
+    }
 
-    observeEvent(ex_dataname, {
-      if (!is.na(ex_dataname)) {
-        output$select_ex_output <- renderUI({
-          checkboxInput(
-            session$ns("select_ex"),
-            "ADEX",
-            value = !is.na(ex_dataname)
-          )
-        })
+    iv <- reactive({
+      iv <- shinyvalidate::InputValidator$new()
+      iv$add_rule("select_ADaM", shinyvalidate::sv_required(
+        message = "At least one ADaM data set is required"
+      ))
+      iv$add_rule("sl_start_date", shinyvalidate::sv_required(
+        message = "Date variable is required"
+      ))
+      if (isTRUE(select_plot()[ex_dataname])) {
+        iv$add_rule("ex_var", shinyvalidate::sv_required(
+          message = "Exposure variable is required"
+        ))
       }
-    })
-
-    observeEvent(rs_dataname, {
-      if (!is.na(rs_dataname)) {
-        output$select_rs_output <- renderUI({
-          checkboxInput(
-            session$ns("select_rs"),
-            "ADRS",
-            value = !is.na(rs_dataname)
-          )
+      if (isTRUE(select_plot()[ae_dataname])) {
+        iv$add_rule("ae_var", shinyvalidate::sv_required(
+          message = "Adverse Event variable is required"
+        ))
+        iv$add_rule("ae_line_var", shinyvalidate::sv_optional())
+        iv$add_rule("ae_line_var", ~ if (length(levels(ADAE[[.]])) > length(ae_line_col_opt)) {
```

Comment:
I don't think this is right as I get:
`Error in levels: object 'ADAE' not found`

Reply:
Works on my end 🤔